### PR TITLE
Add MIT Scene Parsing Benchmark

### PR DESCRIPTION
**Pull request by mariosasko**
Add MIT Scene Parsing Benchmark (a subset of ADE20k).

TODOs:
* [x] add dummy data
* [x] add dataset card
* [x] generate `dataset_info.json`

### CODE REVIEW

**Review by NielsRogge — APPROVED** [review]
Thanks for adding this!

It's interesting that they talk about "scene parsing" and "instance segmentation", as this dataset is used a lot as a benchmark for "semantic segmentation".

File: datasets/scene_parse_150/README.md [hunk 1]
```diff
@@ -0,0 +1,224 @@
...
+- **Leaderboard:** [MIT Scene Parsing Benchmark leaderboard](http://sceneparsing.csail.mit.edu/#:~:text=twice%20per%20week.-,leaderboard,-Organizers)
+- **Point of Contact:** [Bolei Zhou](mailto:bzhou@ie.cuhk.edu.hk)
+
+### Dataset Summary
+
+Scene parsing is to segment and parse an image into different image regions associated with semantic categories, such as sky, road, person, and bed. MIT Scene Parsing Benchmark (SceneParse150) provides a standard training and evaluation platform for the algorithms of scene parsing. The data for this benchmark comes from ADE20K Dataset which contains more than 20K scene-centric images exhaustively annotated with objects and object parts. Specifically, the benchmark is divided into 20K images for training, 2K images for validation, and another batch of held-out images for testing. There are totally 150 semantic categories included for evaluation, which include stuffs like sky, road, grass, and discrete objects like person, car, bed. Note that there are non-uniform distribution of objects occuring in the images, mimicking a more natural object occurrence in daily scene.
```
Comment (lhoestq) [inline datasets/scene_parse_150/README.md:63]: (nits)
```suggestion
Scene parsing is the task of segmenting and parsing an image into different image regions associated with semantic categories, such as sky, road, person, and bed. MIT Scene Parsing Benchmark (SceneParse150) provides a standard training and evaluation platform for the algorithms of scene parsing. The data for this benchmark comes from ADE20K Dataset which contains more than 20K scene-centric images exhaustively annotated with objects and object parts. Specifically, the benchmark is divided into 20K images for training, 2K images for validation, and another batch of held-out images for testing. There are in total 150 semantic categories included for evaluation, which include e.g. sky, road, grass, and discrete objects like person, car, bed. Note that there are non-uniform distribution of objects occuring in the images, mimicking a more natural object occurrence in daily scene.
```

File: datasets/scene_parse_150/README.md [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+}
+```
+
+### Data Fields
+
+#### `scene_parsing`
+
+- `image`: A `PIL.Image.Image` object containing the image. Note that when accessing the image column: `dataset[0]["image"]` the image file is automatically decoded. Decoding of a large number of image files might take a significant amount of time. Thus it is important to first query the sample index before the `"image"` column, *i.e.* `dataset[0]["image"]` should **always** be preferred over `dataset["image"][0]`.
+- `annotation`: A `PIL.Image.Image` object containing the annotation mask.
+- `scene_category`: A scene category for the image.
```
Comment (lhoestq) [inline datasets/scene_parse_150/README.md:109]: Maybe we could have an example of category ?

File: datasets/scene_parse_150/scene_parse_150.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+_SCENE_CATEGORIES = _SCENE_CATEGORIES.strip().split()
+
+
+class SceneParse150(datasets.GeneratorBasedBuilder):
+    """MIT Scene Parsing Benchmark dataset."""
+
+    VERSION = datasets.Version("1.0.0")
+
+    BUILDER_CONFIGS = [
+        datasets.BuilderConfig(name="scene_parsing", version=VERSION, description="The scene parsing variant."),
```
Comment (lhoestq) [inline datasets/scene_parse_150/scene_parse_150.py:192]: should this config be the default one ?

File: datasets/scene_parse_150/README.md [hunk 4]
```diff
@@ -0,0 +1,224 @@
+---
+annotations_creators:
+- crowdsourced
+- expert-generated
+language_creators:
+- found
```
Comment (lhoestq) [inline datasets/scene_parse_150/README.md:6]: well there's no language haha, once the YAML schema is updated we can probably remove that

**Review by lhoestq — APPROVED** [review]
Awesome work, thank you !

File: datasets/scene_parse_150/README.md (hunk 4 above)
Comment (mariosasko) [reply datasets/scene_parse_150/README.md:6]: I use this tag to indicate from where the image data comes. Maybe we can rename it to `data_creators` later?

**Pull request merged**